predicate-builder-service/src/services/application.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Request, Response } from 'express';
+import { PredicateExecutionRequest } from '../classes/predicate-execute-request';
+import { KafkaService } from './kafka.service';
+import { PredicateBuilderService } from './predicate-builder.service';
+
+export class ApplicationService {
+  predicateBuilderService: PredicateBuilderService;
+  kafkaService: KafkaService;
+
+  constructor() {
+    this.predicateBuilderService = new PredicateBuilderService();
+    this.kafkaService = new KafkaService();
+  }
+
+  getOnline(request: Request, response: Response): void {
+    response.send('Predicate Builder Service is online.');
+  }
+
+  executePredicateRequest(
+    request: Request,
+    response: Response,
+    debug = false,
+  ): void {
+    let reqObject: PredicateExecutionRequest | null = null;
```

Comment:
We're declaring this variable without actually using it until after the validation of the request.body has been made. Please declare the variable after the validation of the body has been made so we avoid the `| null = null`

Reply:
It is done this way to catch parse errors and process error individually, it has to be initialized outside of the try-catch block to be used later.